utils/src/test/kotlin/ExtensionsTest.kt

```diff
@@ -45,6 +45,40 @@ class ExtensionsTest : WordSpec({
         }
     }
 
+    "File.isSymbolicLink" should {
+        "correctly detect symbolic links to files on Windows".config(enabled = Os.isWindows) {
+            val dir = createTempDir()
+            dir.resolve("file").createNewFile()
+            ProcessCapture(dir, "cmd", "/c", "mklink", "symlink-to-file", "file")
+            dir.resolve("symlink-to-file").isSymbolicLink() shouldBe true
+            dir.safeDeleteRecursively()
+        }
+
+        "correctly detect symbolic links to directories on Windows".config(enabled = Os.isWindows) {
+            val dir = createTempDir()
+            dir.resolve("directory").safeMkdirs()
+            ProcessCapture(dir, "cmd", "/c", "mklink", "/d", "symlink-to-directory", "directory")
+            dir.resolve("symlink-to-directory").isSymbolicLink() shouldBe true
+            dir.safeDeleteRecursively()
+        }
+
+        "correctly detect hardlinks on Windows".config(enabled = Os.isWindows) {
+            val dir = createTempDir()
+            dir.resolve("file").createNewFile()
+            ProcessCapture(dir, "cmd", "/c", "mklink", "/f", "hardlink", "file")
```

Comment:
Is a `hardlink` actually a `symbolic link` on windows?

Reply:
Ah, good point, no, it's a hardlink as also known from Linux. Let's see how the tests behave on AppVeyor, it should indeed fail. I currently cannot test this locally.

Reply:
My understanding of a `hardlink` under Linux is that roughly speaking you cannot distinguish between different hardlinks to the same file. That is a hardlink is just a reference from a file system (directory) node and a file exists as long as there is at least one `hardlink` pointing to it. 

I guess terminoligy of `hardlink` under Windows is analog - thus we'd just need to flip the `asserts expected value` ?!

Reply:
While we could do that, I think it's clearer to just drop that test case.

Reply:
Changed my mind, will even add more tests :-)